### PR TITLE
Support using direct ufs uri when dora is disabled

### PR DESCRIPTION
**Pull request by Jackson-Wang-7**
### What changes are proposed in this pull request?
Implement some functions when Dora is disabled, so the client can access UFS directly without dora cluster.

### Why are the changes needed?
Support the alluxio client using UFS URI to access UFS directly

### CODE REVIEW

File: dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java [hunk 1]
```diff
@@ -151,4 +157,37 @@ protected AlluxioURI getAlluxioPath(Path path) {
   protected Path getFsPath(String fsUriHeader, URIStatus fileStatus) {
     return new Path(fsUriHeader + fileStatus.getPath());
   }
+
+  @Override
+  public BlockLocation[] getFileBlockLocations(FileStatus file, long start, long len)
+      throws IOException {
+    if (mFileSystem.getConf().getBoolean(PropertyKey.DORA_ENABLED)) {
```
Comment (beinan) [inline dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java:164]: Is it possible to make this value as a field of this class?  then we can initialize the value in the constructor or passed in from caller,  becuase reading a value from conf for every read would be expensive.

File: dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java [hunk 2]
```diff
@@ -151,4 +157,37 @@ protected AlluxioURI getAlluxioPath(Path path) {
...
+      return super.getFileBlockLocations(file, start, len);
+    } else {
+      // If Dora is not enabled, use the localhost as the default block location.
```
Comment (beinan) [inline dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java:167]: hmmmm,  I think dora will be always enabled in alluxio 3xx

File: dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java (hunk 1 above)
Comment (beinan) [reply dora/core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java:164]: I don't think we need the check.  dora will be always enabled

File: dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java [hunk 3]
```diff
@@ -617,6 +620,21 @@ public synchronized boolean getUriValidationEnabled() {
     return mUriValidationEnabled;
   }
 
+  /**
+   * Sets current uri path with given uri path.
+   * @param uriPath
+   */
+  public void setUriPath(String uriPath) {
```
Comment (beinan) [inline dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java:627]: I'm not a big fan of passing the value through FileSystemContext, I prefer to explicitly pass the value of ufs path

Comment (Jackson-Wang-7) [reply dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java:627]: I will consider this

**Comment by huanghua78** [comment]
What do you mean by "Dora is disabled"?
I don't think Dora can be disabled.
Please clarify this with more details.

**Comment by Jackson-Wang-7** [comment]
@huanghua78 @beinan When you want to use alluxio client to access UFS directly and dora cluster is not involved in, we can use `dora.enabled=false` to bypass DoraCacheFileSystem. That's what we want to do in only-localcache sdk case and the related code change are in this PR. If you want to combine the client cache and system cache, you can still choose `alluxio.user.client.cache.enabled=true` and `alluxio.dora.enabled=true`.

**Comment by huanghua78** [comment]
> @huanghua78 @beinan When you want to use alluxio client to access UFS directly and dora cluster is not involved in, we can use `dora.enabled=false` to bypass DoraCacheFileSystem. That's what we want to do in only-localcache sdk case and the related code change are in this PR. If you want to combine the client cache and system cache, you can still choose `alluxio.user.client.cache.enabled=true` and `alluxio.dora.enabled=true`.

I see. Thanks for this explanation.

**Pull request closed without merging**